### DISCUSSION TITLE
Ignoring #### Empty Tracks and Empty CSVs

### DISCUSSION THREAD

**Discussion by BKTidswell** [talmolab/sleap#1674] [Help!] [answered]
Hello,
I am trying to export some of my predictions as csvs, but for about a third of the videos I get the error "Ignoring #### Empty Tracks" (with the number around 3200), and it produces a CSV with no data in it. When I look at the videos I see tracks, and I only see the 8 tracks that I limited it to. All of the predictions were created at the same time, so I'm not sure why some of them are causing this issue. Happy to provide any information to clear this up, and I've included screenshots of a representative video.

<img width="1800" alt="Screenshot 2024-01-29 at 4 36 20 PM" src="https://github.com/talmolab/sleap/assets/21346180/38130f6c-a53f-422a-9506-462f4925cc71">
<img width="1799" alt="Screenshot 2024-01-29 at 4 36 28 PM" src="https://github.com/talmolab/sleap/assets/21346180/3b31bb88-7e78-4998-94e1-c5e42c089fd5">

**Comment by roomrys** [comment]
Hi @BKTidswell,

From the screenshots you sent, I also see plenty of `Tracks` both on the seekbar and on the Instances dock you opened for us. 

I am wondering if we are looking at the correct video and whether it is possible there are multiple videos with the same name (albeit different path). A useful piece of information would be the copy/pasted terminal message starting from `Happy SLEAPing`.

Thanks,
Liezl

**Reply by BKTidswell** [reply]
Hello Liezl, 
Thanks for the quick response! Here's another video as an example, as well as what I see in the terminal after running "Export Analysis CSV"

All the videos are in the same directory, so they can't have the same name.

<img width="1789" alt="Screenshot 2024-01-29 at 4 48 48 PM" src="https://github.com/talmolab/sleap/assets/21346180/82acdef9-e6ce-4464-aea9-9f71cac9e3aa">

<img width="795" alt="Screenshot 2024-01-29 at 4 48 33 PM" src="https://github.com/talmolab/sleap/assets/21346180/6aceabac-fa72-44fb-b601-d9a7413a726a">

`Happy SLEAPing! :)
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3185 empty tracks
ignoring 3184 empty tracks
ignoring 3193 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3186 empty tracks
ignoring 3193 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3193 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3184 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3193 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3193 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3185 empty tracks
ignoring 3192 empty tracks
ignoring 3185 empty tracks
ignoring 3193 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3200 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3185 empty tracks
ignoring 3184 empty tracks
ignoring 3184 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3192 empty tracks
ignoring 3184 empty tracks
ignoring 3184 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3200 empty tracks
ignoring 3185 empty tracks`

**Comment by BKTidswell** [comment, marked as answer]
When I deleted all unused tracks it brought the number down to around 1920 unused tracks for each video, and now I no longer have any 0 kb CSVs produced. I'm not sure why this was an issue before with 3200 unused tracks or how that fixed it, but I wanted to put this here for the record.

**Reply by roomrys** [reply]
Thanks for the follow-up. 

Ok, let's see if we can make sense of the result. Since `Track`s are limited to the scope of a single `Video` and you have at least 8 `Track`s per `Video` (ideally 1 `Track` per animal) and 160 `Video`s, I would expect there to be a minimum of 160 * 8 = 1280 `Track`s - which is at least in the ball park range of 1920 (1920 `Track`s / 160 `Video`s = 12 `Track`s per `Video`).

The analysis file is exported per `Video` and would skip over `Track`s that are not used in the `Video` being exported - hence the messages.

As far as the 0kB CSVs... I know the analysis file first creates a matrix with one dimension being the size of all `Track`s in the project and then uses an occupancy mask to slim down the matrix to which `Track`s were actually used. So, I wonder if the number of `Track`s in the entire project (although unused `Track`s) was too large to allocate space for this matrix.

Just a hypothesis that would need to be tested/confirmed.

Liezl
